internal/gatewayapi/resource/load.go

```diff
@@ -117,6 +141,21 @@ func loadKubernetesYAMLToResources(input []byte, addMissingResources bool) (*Res
 		data := kobjVal.FieldByName("Data")
 		stringData := kobjVal.FieldByName("StringData")
 
+		// Check if this resource is managed by the ExtensionManager and if so, classify it
+		if len(extGVKMap) > 0 {
+			key := fmt.Sprintf("%s/%s/%s", gvk.Group, gvk.Version, gvk.Kind)
+			if category, ok := extGVKMap[key]; ok {
+				un.SetNamespace(namespace)
+				switch category {
+				case extFilter, extBackend:
+					resources.ExtensionRefFilters = append(resources.ExtensionRefFilters, *un)
```

Comment:
is this logic right of adding the ext backend into ext ref flters ?
cc @Xunzhuo